File: core/src/main/java/net/jradius/client/RadiusMultiClient.java
package net.jradius.client;

import java.io.IOException;
import java.net.DatagramSocket;
import java.net.InetAddress;
import java.net.SocketException;
import org.ehcache.Cache;
import org.ehcache.CacheManager;

public class RadiusMultiClient extends RadiusClient 
{
	CacheManager cacheManager;
	String requestCacheName;
	Cache<java.io.Serializable, Object> requestCache;
	
	public RadiusMultiClient() throws IOException 
	{
		super();
	}

	public RadiusMultiClient(DatagramSocket socket, InetAddress address, String secret, int authPort, int acctPort, int timeout) throws SocketException 
	{
		super(socket, address, secret, authPort, acctPort, timeout);
	}

	public RadiusMultiClient(DatagramSocket socket, InetAddress address, String secret) 
	{
		super(socket, address, secret);
	}

	public RadiusMultiClient(InetAddress address, String secret, int authPort, int acctPort, int timeout) throws IOException 
	{
		super(address, secret, authPort, acctPort, timeout);
	}

	public RadiusMultiClient(InetAddress address, String secret) throws IOException 
	{
		super(address, secret);
	}

	public RadiusMultiClient(RadiusClientTransport transport) 
	{
		super(transport);
	}
}

File: client-dotnet/RadiusMultiClient.cs
using Microsoft.Extensions.Caching.Memory;
using System.IO;
using System.Net;
using System.Net.Sockets;

namespace JRadius.Core.Client
{
    public class RadiusMultiClient : RadiusClient
    {
        private IMemoryCache _requestCache;

        public RadiusMultiClient(IMemoryCache requestCache)
            : base()
        {
            _requestCache = requestCache;
        }

        public RadiusMultiClient(IMemoryCache requestCache, UdpClient socket, IPAddress address, string secret, int authPort, int acctPort, int timeout)
            : base(socket, address, secret, authPort, acctPort, timeout)
        {
            _requestCache = requestCache;
        }

        public RadiusMultiClient(IMemoryCache requestCache, UdpClient socket, IPAddress address, string secret)
            : base(socket, address, secret)
        {
            _requestCache = requestCache;
        }

        public RadiusMultiClient(IMemoryCache requestCache, IPAddress address, string secret, int authPort, int acctPort, int timeout)
            : base(address, secret, authPort, acctPort, timeout)
        {
            _requestCache = requestCache;
        }

        public RadiusMultiClient(IMemoryCache requestCache, IPAddress address, string secret)
            : base(address, secret)
        {
            _requestCache = requestCache;
        }

        public RadiusMultiClient(IMemoryCache requestCache, IRadiusClientTransport transport)
            : base(transport)
        {
            _requestCache = requestCache;
        }
    }
}
